lib/Core/Site/Values/Fields.php

```diff
@@ -236,6 +236,45 @@ public function getFieldById($id): APIField
         return $this->getSurrogateField((string) $id, $this->content);
     }
 
+    /**
+     * {@inheritdoc}
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     */
+    public function getFirstSetField(string $firstIdentifier, string ...$otherIdentifiers): APIField
+    {
+        $identifiers = \array_merge([$firstIdentifier], $otherIdentifiers);
+        $fields = $this->getAvailableFields($identifiers);
+
+        foreach ($fields as $field) {
+            if (!$field->isEmpty()) {
+                return $field;
+            }
+        }
+
+        return $fields[0] ?? $this->getSurrogateField($firstIdentifier, $this->content);
```

Comment:
Why do we need `$fields[0]` here?

So, if we have 5 fields, first we return first non-empty one, but if there are no empty ones, we return first field even if it is empty. Sounds too much complicated.

Reply:
The reason is I wanted to return surrogate field only if no fields were found. But I agree, it is a bit complicated.

Reply:
Changed in https://github.com/netgen/ezplatform-site-api/pull/150/commits/c76c33d49e6c1655b6b990ef7899cc9d549fcc42

Reply:
Restored returning first empty field, as it might be useful if for example using field definition in some way.